pkg/asset/installconfig/alibabacloud/validation.go

```diff
@@ -172,6 +173,7 @@ func validatePrivateZoneID(client *Client, ic *types.InstallConfig, path *field.
 // ValidateForProvisioning validates if the install config is valid for provisioning the cluster.
 func ValidateForProvisioning(client *Client, ic *types.InstallConfig, metadata *Metadata) error {
 	allErrs := field.ErrorList{}
+	allErrs = append(allErrs, validateMetadataServerIPNotInMachineCIDR(ic.Networking)...)
```

Comment:
Because these functions are static validation--that is, they do not require connecting to the alibaba api--they should be moved to `pkg/types`. Specifically here: https://github.com/openshift/installer/blob/master/pkg/types/validation/installconfig.go#L465

the `validatePlatform` function will need to pass in the networking struct, similar to how OpenStack does.